src/Orleans/Statistics/CounterStatistic.cs

```diff
@@ -103,9 +110,8 @@ internal static void SetOrleansManagedThread()
             {
                 lock (lockable)
                 {
+                    allValuesFromSpecificThread = new List<ReferenceLong>(new ReferenceLong[500]);
```

Comment:
I would use a smaller initial value, like 100, and when growing use valuesToAdd  of 100 too.
And use constant for this value.